presto-main/src/main/java/io/prestosql/operator/scalar/BitwiseFunctions.java

```diff
@@ -75,4 +75,55 @@ public static long bitwiseXor(@SqlType(StandardTypes.BIGINT) long left, @SqlType
     {
         return left ^ right;
     }
+
+    @Description("bitwise left shift")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitwiseLsh(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.BIGINT) long shift, @SqlType(StandardTypes.BIGINT) long digits)
+    {
+        if (digits < 0 || digits > 64) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Digits specified in bitwise_lsh must be between 0 and 64, got " + digits);
+        }
+        if (digits == 64) {
+            return num << shift;
+        }
+        else {
+            long mask = (1 << digits) - 1;
+            return (num << shift) & mask;
+        }
+    }
+
+    @Description("bitwise logical right shift")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitwiseLrsh(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.BIGINT) long shift, @SqlType(StandardTypes.BIGINT) long digits)
+    {
+        if (digits < 0 || digits > 64) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Digits specified in bitwise_lrsh must be between 0 and 64, got " + digits);
+        }
+        if (digits == 64) {
+            return num >>> shift;
+        }
+        else {
+            long mask = (1 << digits) - 1;
+            return (num & mask) >>> shift;
+        }
+    }
+
+    @Description("bitwise arithmetic right shift")
+    @ScalarFunction
+    @SqlType(StandardTypes.BIGINT)
+    public static long bitwiseArsh(@SqlType(StandardTypes.BIGINT) long num, @SqlType(StandardTypes.BIGINT) long shift, @SqlType(StandardTypes.BIGINT) long digits)
+    {
+        if (digits < 0 || digits > 64) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Digits specified in bitwise_arsh must be between 0 and 64, got " + digits);
+        }
+        if (digits == 64) {
+            return num >> shift;
+        }
+        else {
+            long mask = digits == 64 ? Long.MAX_VALUE : (1 << digits) - 1;
+            return (num & mask) >> shift;
```

Comment:
`digits == 64` is always false because the condition is already returned at L122.

In my opinion, you don't need to use `else` like `bitcount` function.
```java
        if (digits == 64) {
            return num >> shift;
        }
        long mask = (1 << digits) - 1;
        return (num & mask) >> shift;
```

Reply:
Ah, true. I will update. Thanks.